Cleans up page element alignments and makes the page more mobile friendly

jasoryeh opened the pull request:
Just a bit of changes to make the page more aligned so the images aren't flowing off the page on mobile, and so there isn't a random hovering bar when you first load the page.

- Clean up *slightly* misaligned elements
- Adds the nav-bar to config editor
- Makes page more mobile friendly with more media queries
- Makes navbar completely transparent when not scrolled, and adds Geyser logo when scrolled

jasoryeh committed:
Cleans up page element alignments and makes the page more mobile friendly
- Clean up *slightly* misaligned elements
- Adds the nav-bar to config editor
- Makes page more mobile friendly with more media queries
- Makes navbar completely transparent when not scrolled, and adds Geyser logo when scrolled

rtm516 commented on config_editor.html:
```diff
@@ -7,9 +7,17 @@
     <link rel='stylesheet' type='text/css' media='screen' href='css/config-editor.css'>
+    <link rel="stylesheet" href="css/style.css" />
+    <style>
+        nav * {
+            color: black !important;
+        }
```
Can this be pulled into the `css/config-editor.css` file?

rtm516 commented on css/style.css:
```diff
@@ -51,19 +79,31 @@ h1.text-center {
 	background-repeat: no-repeat;
 }
 
+.sections .col {
+	padding: 80px 30px 80px 10px;
+}
+.sections img {
+	width: 100%;
+}
+
+@media screen and (max-width: 768px) {
+	.sections .col {
+		padding: 20px 20px 20px 20px;
+		width: 100%;
+	}
+	.sections img {
+		width: 95vw;
+	}
+}
+
 .section2 {
 	background-color: #212121;
 	min-width: 400px;
 }
 
-.section2 .col {
-	padding: 80px 30px 30px 10px;
-}
+.section2 .col {}
 
-.section2 img {
-	height: 200px;
-	width: 400px;
-}
+.section2 img {}
```
These can be removed

rtm516 commented on css/style.css:
```diff
@@ -92,13 +131,9 @@ h1.text-center {
-	max-width: 600px;
-	width: 97vw;
-}
+.section4 img {}
```
As above

rtm516 commented on index.html:
```diff
@@ -195,18 +187,26 @@ <h6 class="mb-0">&copy 2020 GeyserMC</h6>
                   $(".navbar").css("background-color", "rgba(0, 0, 0, 0.9)");
+                  $(".navbar").css("box-shadow", "0 .5rem 1rem rgba(0, 0, 0, 0.15)");
                   $(".navbar").css("margin", "0px");
-                  $(".navbar").css("padding", "20px 20px 10px 20px");
+                  $(".navbar").css("padding", "10px 10px 10px 10px");
+                  $(".navbar").css("border-radius", "0");
+                  $('.nav-logoimg').css("opacity", '1.0');
                 } else {
-                  $(".navbar").css("background-color", "rgba(255, 255, 255, 0.1)");
+                  $(".navbar").css("background-color", "rgba(255, 255, 255, 0.0)");
+                  $(".navbar").css("box-shadow", "0 .5rem 1rem rgba(0, 0, 0, 0.0)");
                   $(".navbar").css("margin", "10px");
-                  $(".navbar").css("padding", "10px 10px 10px 10px");
+                  $(".navbar").css("padding", "5px 10px 5px 10px");
+                  $(".navbar").css("border-radius", "25px");
+                  $('.nav-logoimg').css("opacity", '0.0');
```
Can `.navbar` and `.nav-logoimg` be pulled into variables to prevent searching for the element every time?

rtm516 commented:
There are a few issues caused by this PR
 - The homepage on a 1920x1080 device isn't full width so leaves a white gap on the right-hand side
 - The buttons on the config editor page
  - Don't line up
  - Text between is invisible on the background as its white on white
 - The navigation on the config editor page
   - Has no hover effects on the buttons
   - Isn't full width of the page
 - The sections in the config editor now have rounded corners so would look better with gaps in-between.
 - The navbar on the homepage
   - Drifts up slightly on load
   - Moves when you start scrolling

jasoryeh committed:
Use suggested changes from @rtm516

jasoryeh committed:
Fixes right side whitebars as a result of a7de093

jasoryeh committed:
Properly align config editor buttons.

jasoryeh committed:
Fix nav animations on config-editor and main page

jasoryeh committed:
Fix modified logo while fixing onload navbar movement

jasoryeh commented:
> There are a few issues caused by this PR
> 
> * The homepage on a 1920x1080 device isn't full width so leaves a white gap on the right-hand side ✔
> * The buttons on the config editor page don't line up ✔
> * Text between is invisible on the background as its white on white ✔
> * The navigation on the config editor page
>   * Has no hover effects on the buttons ✔
>   * Isn't full width of the page ❓
> * The sections in the config editor now have rounded corners so would look better with gaps in-between. ✔
> * The navbar on the homepage 
>   * Drifts up slightly on load ✔
>   * Moves when you start scrolling 💯This is kind of purposeful, to make the navbar take a not so giant chunk of screen real estate. I edited the animations to make the move and shrink smoother.

Fixed most of those issues, and fixed some missing things while I was at it.

jasoryeh committed:
Correct modified nav dimensions when scrolled

rtm516 commented on config_editor.html:
```diff
@@ -24,9 +27,52 @@
+        <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav"
+          aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
+          <span class="navbar-toggler-icon"></span>
+        </button>
+        <div class="collapse navbar-collapse justify-content-end" id="navbarNav">
+            <ul class="navbar-nav">
+                <li class="nav-item">
+                    <a href="http://ci.geysermc.org">
+                        <button class="btn nav-link">
+                            <i class="fas fa-cloud-download-alt"></i>&nbsp;<span>Download</span>
+                        </button>
+                    </a>
+                </li>
+                <li class="nav-item">
+                    <a class="nav-link" href="https://github.com/GeyserMC/Geyser">
+                        <i class="fab fa-github"></i>&nbsp;<span>Source</span>
+                    </a>
+                </li>
+                <li class="nav-item">
+                    <a class="nav-link" href="http://discord.geysermc.org">
+                        <i class="fab fa-discord"></i>&nbsp;<span>Discord</span>
+                    </a>
+                </li>
+                <li class="nav-item">
+                    <a class="nav-link" href="https://github.com/GeyserMC/Geyser/wiki/Setup">
+                        <i class="fas fa-tools"></i>&nbsp;<span>Setup</span>
+                    </a>
+                </li>
+                <li class="nav-item">
+                    <a class="nav-link" href="config_editor.html">
+                        <i class="fas fa-sliders-h"></i>&nbsp;<span>Config Editor</span>
+                    </a>
+                </li>
+                <!--
+                <li class="nav-item">
+                    <a class="nav-link" href="https://patreon.com/GeyserMC"><i class="fab fa-patreon"></i> Donate</a>
+                </li>
+                -->
+            </ul>
+        </div>
```
Could this be wrapped in a `<div class="container">` to pull the logo and buttons in so it doesn't look as strange with the rest of the page?

jasoryeh replied:
Unfortunately doing so will throw off things inside the nav, and things start to move to the next line

jasoryeh replied:
Gotta play around with it, but the other css we had was sort of wonky already so a lot of the changes I proposed were trying to fix up the weird styling.

jasoryeh replied:
Resolved with invert when mobile

rtm516 commented on config_editor.html:
```diff
@@ -24,9 +27,52 @@
     <meta name="theme-color" content="#ffffff">
 </head>
 <body>
+    <nav class="navbar navbar-expand-lg navbar-dark fixed-top">
```
This doesn't have a mobile navigation bar making it impossible to go anywhere else but 'back' on mobile

jasoryeh replied:
Resolved with grey background for mobile users.

rtm516 reviewed (changes requested):
Nice work so far, but along with these few minor last changes could you:
 - Add a background to the mobile nav when on the first slide on the homepage
 - Look into the horizontal scrollbar sometimes visible on mobile device sizes, looks like it may be caused by the wider images not being scaled.

jasoryeh committed:
Add background on nav for mobile users on config editor.

jasoryeh committed:
Fixes invisible menu toggler when on mobile in config editor
